plugins/inputs/logfile/fileconfig.go

```diff
@@ -144,6 +174,24 @@ func (config *FileConfig) init() error {
 		config.RetentionInDays = -1
 	}
 
+	if config.Filters != nil {
+		filters := make([]LogFilter, len(config.Filters))
+		for i, f := range config.Filters {
+			err = f.init()
+			if err != nil {
+				return err
```

Comment:
`return` on error instead of `continue` means that the entire `logfile` plugin will not get started.
```
	// Initialize all the file configs
	for i := range t.FileConfig {
		if err := t.FileConfig[i].init(); err != nil {
			return err
		}
	}

	t.started = true
```
So no files will be monitored.

I think you made the right choice.
CWA should fail quickly and loudly so the user knows there is a problem and can fix their configuration.

Just thought I'd point this out in case others think CWA should just log the bad filter and continue to monitor this file using any other valid filters.

Reply:
> CWA should fail quickly and loudly so the user knows there is a problem

Yup, we're on the same page for this. There's a difference between writing an improper expression (something that doesn't do what you expect) versus writing an invalid one (one that doesn't compile). If the customer chooses to use the expression and write an expression that doesn't filter out what they want, that's not something we can control. But we CAN control the behavior if they supply a regular expression that has invalid syntax.